pkg/network/protocols/kafka/stats.go

```diff
@@ -66,24 +91,72 @@ func (r *RequestStats) CombineWith(newStats *RequestStats) {
 			// Nothing to do in this case
 			continue
 		}
-		r.AddRequest(statusCode, newRequests.Count, newRequests.StaticTags)
+
+		if newRequests.Latencies == nil {
+			// In this case, newRequests must have only FirstLatencySample, so use it when adding the request
+			r.AddRequest(statusCode, newRequests.Count, newRequests.StaticTags, newRequests.FirstLatencySample)
+			continue
+		}
```

Comment:
where and how do you ensure it runs only for fetch requests?

Reply:
What’s the issue with it running for produce requests? The latency will always be 0 in this case. Additionally, on the backend, we're currently only processing latency for fetch requests